worker/caasprober/probe.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasprober
+
+import (
+	"github.com/juju/juju/observability/probe"
+)
+
+// CaasProbes provides a private internal implementation of CAASProbes.
+type CAASProbes struct {
```

Comment:
if this is a private implementation, it shouldn't be exported
same with the constructor